lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -88,6 +88,43 @@ LinearLayout identityND(StringAttr inDimName, ArrayRef<unsigned> shape,
   return ret;
 }
 
+// Returns 2D layout that traverses the 2nd leading dimension first and then
+// 1st leading dimension. Only supports 2D layout for now.
+LinearLayout transposeND(StringAttr inDimName, ArrayRef<unsigned> shape,
```

Comment:
Do you mean merging this into `identityND` with an additional flag to construct the LL, instead of creating a new function? I think I've tried that before and the problem was the conflict with basis dim StringAttr, but I can take another look.

Reply:
> Do you mean merging this into identityND with an additional flag to construct the LL, instead of creating a new function?

Yes, I think this is cleaner. You can take another look first and we can check with Lei later.

Reply:
`transposeND` cannot be merged with `identityND` because simply reversing `identityND` will change the leading dimension from N-dim to K-dim. At the same time, reversing `identityND` will still keep it as identity matrix, whereas the correct is not identity anymore, therefore this function needs to be standalone.